bluemira/equilibria/fem_fixed_boundary/utilities.py

```diff
@@ -210,6 +210,80 @@ def get_tricontours(x, z, array, value):
     return contours
 
 
+def find_flux_surface(psi_norm_func, psi_norm, mesh=None, n_points=100):
+    """
+    Find a flux surface in the psi_norm function precisely by normalised psi value.
+
+    Parameters
+    ----------
+    psi_norm_func: Callable
+        Function to calculate normalised psi
+    psi_norm: float
+        Normalised psi value for which to find the flux surface
+    mesh: Optional[dolfin.Mesh]
+        Mesh object to use to estimate the flux surface
+        If None, reasonable guesses are used.
+    n_points: int
+        Number of points along the flux surface
+
+    Returns
+    -------
+    points: np.ndarray
+        x, z coordinates of the flux surface
+    """
+    x_axis, z_axis = find_magnetic_axis(lambda x: -psi_norm_func(x), mesh=mesh)
+
+    if mesh:
+        search_range = mesh.hmax()
+        mpoints = mesh.coordinates()
+        psi_norm_array = [psi_norm_func(x) for x in mpoints]
+        contour = get_tricontours(
+            mpoints[:, 0], mpoints[:, 1], psi_norm_array, psi_norm
+        )[0]
+        d_guess = abs(np.max(contour[0, :]) - x_axis) - search_range
+        bounds = [(d_guess - 2 * search_range, d_guess + 2 * search_range)]
+
+        def get_next_guess(points, distances, i):
+            return np.hypot(points[0, i - 1] - x_axis, points[1, i - 1] - z_axis)
+
+    else:
+        d_guess = 0.5
+        bounds = [(0.1, None)]
+
+        def get_next_guess(points, distances, i):
```

Comment:
Looks like `points` isn't used here

Reply:
Yes this is because of the rather weird requirement to have a different function for different cases, which take different arguments. So I just overloaded them both with different unused arguments. I've added a noqa but happy to improve if you have a better approach!

Reply:
I don't have a better option at the moment. Merging and we may come back to it in the future